Sources/ApolloCodegenLib/ApolloCodegen.swift

```diff
@@ -212,6 +218,24 @@ public class ApolloCodegen {
       throw Error.schemaNameConflict(name: context.schemaNamespace)
     }
   }
+  
+  /// Validates that there are no type conflicts within a SelectionSet
+  static private func validateTypeConflicts(for selectionSet: IR.SelectionSet, with context: ConfigurationContext) throws {
+    // Check for type conflicts resulting from singularization/pluralization of fields
+    var typeNames = [String: String]()
```

Comment:
This is a little hard to understand. It's not clear what the key vs value represent here. Maybe changing this to `fieldNamesByFormattedTypeName` or at least adding an inline documentation comment to clarify?